identity-access-manager-keycloak/docker-compose-postgres.cluster.yml

```diff
@@ -0,0 +1,75 @@
+version: '3.9'
+
+services:
+  keycloak-postgres-1:
+    environment:
+      REPMGR_PARTNER_NODES: ${KC_REPMGR_PARTNER_NODES}
+    deploy:
+      placement:
+        constraints:
+          - "node.labels.name==node-1"
+
+  keycloak-postgres-2:
+    image: bitnami/postgresql-repmgr:14
+    environment:
+      POSTGRESQL_PASSWORD: ${KC_POSTGRESQL_PASSWORD}
+      POSTGRESQL_USERNAME: ${KC_POSTGRESQL_USERNAME}
+      POSTGRESQL_DATABASE: ${KC_POSTGRESQL_DATABASE}
+      REPMGR_NODE_NETWORK_NAME: keycloak-postgres-2
+      REPMGR_PASSWORD: ${KC_REPMGR_PASSWORD}
+      REPMGR_RECONNECT_INTERVAL: 3
+      REPMGR_NODE_NAME: keycloak-postgres-2
+      REPMGR_PRIMARY_HOST: ${KC_REPMGR_PRIMARY_HOST}
+      REPMGR_PARTNER_NODES: ${KC_REPMGR_PARTNER_NODES}
+    volumes:
+      - 'keycloak-postgres-2-data:/bitnami/postgresql'
+    deploy:
+      placement:
+        constraints:
+          - "node.labels.name==node-2"
+      replicas: 1
+      resources:
+        limits:
+          cpus: ${KC_POSTGRES_CPU_LIMIT}
+          memory: ${KC_POSTGRES_MEMORY_LIMIT}
+        reservations:
+          cpus: ${KC_POSTGRES_CPU_RESERVE}
+          memory: ${KC_POSTGRES_MEMORY_RESERVE}
+    networks:
+      default:
+      keycloak_backup_net: {}
```

Comment:
do we need to backup data of keycloak ?

Reply:
Hmm, we probably don't need the active sessions to be backed up (worst case, people will need to login again), but I think it would be important to back up the user data, so if more users (not just the single test user we have now) are registered on the platform, they will need to have their accounts restored if there is a system wide crash of keycloak. So yes, we do need this. The session data might also be useful for auditing purposes.